src/main/java/hunternif/mc/impl/atlas/mixin/MixinCartographyContainer.java

```diff
@@ -0,0 +1,95 @@
+package hunternif.mc.impl.atlas.mixin;
+
+import hunternif.mc.impl.atlas.api.AtlasAPI;
+import hunternif.mc.impl.atlas.mixinhooks.CartographyTableHooks;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.inventory.CraftResultInventory;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.inventory.container.CartographyContainer;
+import net.minecraft.inventory.container.Container;
+import net.minecraft.inventory.container.ContainerType;
+import net.minecraft.inventory.container.Slot;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.util.IWorldPosCallable;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import javax.annotation.Nullable;
+
+//TODO Verify if the methods in this yarn methods in this class actually equal the MCP versions with Linkie
+@Mixin(targets = "net.minecraft.inventory.container.CartographyContainer$4")
```

Comment:
These mixins only worked in dev environment for me. I guess it will break on forge to.
See https://github.com/AntiqueAtlasTeam/AntiqueAtlas/commit/a5614b5c4d7a80148d4f3c7987e175ca265b935b

Reply:
Apparently, because they try to mixin into an anonymous class, somehow the mapping doesn't work

Reply:
Yes, you get the same issue when loading the mod outside of a dev env:

> [20:10:59] [pool-3-thread-1/FATAL] [mixin/]: Mixin apply failed antiqueatlas.mixins.json:MixinCartographyContainerSlot -> net.minecraft.inventory.container.CartographyContainer$4: org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException Critical injection failure: @ Inject annotation on antiqueatlas_canInsert could not find any targets matching 'isItemValid' in net.minecraft.inventory.container.CartographyContainer$4.